### PR TITLE
Connecting up newsletter sign up

### PR DESCRIPTION
**Pull request by AndrewBestbier**
Another PR guys! Are you excited?
Alt Text

Okay. How about if I show you a cute animal?
Alt Text

Here you go:
Alt Text

### Motivation
This ticket has to do with actually making the Mailing List component work.

### Test plan

- Get a MailChimp account and view our test list
- Sign up using no email address and check that an error is shown
- Sign up using an invalid email address and check that an error is shown
- Sign up with an email address and view the second component

### Pre-merge checklist

- [x] Unit tests
- [x] Reviews
  - [x] Code review
  - [x] Design review
- [x] Manual testing
  - [x] Windows 7 IE 11
  - [x] Windows 10 Edge
  - [x] Mac OS El Capitan Safari
  - [x] Mac OS El Capitan Chrome
  - [x] Mac OS El Capitan Firefox
  - [x] iOS 9 Safari
  - [x] Android 6 Chrome
- [ ] Tester approved

### CODE REVIEW

**Comment by zoeabryant** [comment]
error message for if you've already signed up

File: services/mailchimp/utilities/index.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        'Content-Type': 'application/json',
+        Authorization: 'Basic ' + new Buffer(username + ':' + apiKey).toString('base64'),
+      },
+      mode: 'cors',
+      body,
+    })
+    .then(response => response.json());
+}
+
+export function formatResponse(res, data) {
```
Comment (zoeabryant) [inline services/mailchimp/utilities/index.js]: formatPostReponse formatPatchReponse

File: services/mailchimp/utilities/index.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
+/* eslint-disable no-console */
+import fetch from 'node-fetch';
+import crypto from 'crypto';
+
+const algorithm = 'aes256';
+const key = process.env.SECRET_ENCRYPTION_KEY || 'secret';
```
Comment (zoeabryant) [inline services/mailchimp/utilities/index.js]: This was a workaround for a test, set in test

File: services/mailchimp/utilities/index.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+export function encryptText(text) {
```
Comment (zoeabryant) [inline services/mailchimp/utilities/index.js]: encrtypText(text, k = getKey())

File: services/mailchimp/utilities/index.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    email_address: encryptedEmail,
+  };
+}
+
+
+export function formatFormInput(event, isEmailEncrypted) {
+  let fullName = [];
+  let emailAddress = event.body.email_address;
+  if (event.body.name) {
+    fullName = event.body.name.split(' ');
```
Comment (zoeabryant) [inline services/mailchimp/utilities/index.js]: Look at the names issue

Comment (zoeabryant) [inline services/mailchimp/utilities/index.js]: Split front end name and surname

File: services/mailchimp/utilities/test.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+    const result = formatResponse(test, data);
+    expect(result).to.deep.equal({
+      newsletterSubmitted: true,
+      errorMessage: '',
+      email_address: encryptText('test@gmail.com'),
+      updatedFormSubmitted: false,
+    });
+  });
+  it('returns the correct values if there are no erros and a new account has been created', () => {
```
Comment (zoeabryant) [inline services/mailchimp/utilities/test.js]: errors

File: site/css/_colors.css [hunk 6]
```diff
@@ -12,6 +12,7 @@
 @value red: #FC1D42;
 @value white: #F8F8F8;
 @value black: #212121;
+@value dark-red: #bb1e38;
```
Comment (zoeabryant) [inline site/css/_colors.css]: remove

File: site/pages/home/newsletter-slice/after-sign-up/index.js [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    this.triggerReflow();
+  }
+
+  componentWillReceiveProps() {
+    const newState = {};
+    newState.submitting = false;
+    this.setState(newState);
+  }
+
+  handleInputChange(event) {
```
Comment (zoeabryant) [inline site/pages/home/newsletter-slice/after-sign-up/index.js]: handleInputChange = (event) => {

File: site/pages/home/newsletter-slice/after-sign-up/index.js [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  componentDidMount() {
+    this.triggerReflow();
+  }
+
+  componentWillReceiveProps() {
+    const newState = {};
```
Comment (zoeabryant) [inline site/pages/home/newsletter-slice/after-sign-up/index.js]: one liner

File: site/pages/home/newsletter-slice/after-sign-up/index.js [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+            <form className={styles.form}>
+              <div className={styles.formBlock}>
+                <label htmlFor="name" className={styles.formLabel}>Full name</label>
+                <input
+                  id="name"
+                  name="name"
+                  type="text" placeholder="John Smith"
+                  className={!this.props.errorMessage ?
+                    styles.formInput :
+                    cx('formInput', 'afterFormError')
```
Comment (zoeabryant) [inline site/pages/home/newsletter-slice/after-sign-up/index.js]: https://github.com/redbadger/website-honestly/blob/master/site/pages/home/contact-us-slice/form/index.js#L42

File: site/pages/home/newsletter-slice/after-sign-up/index.js [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+  }
+
+  render() {
+    return (
+      <section className={cx('newsletter', 'submitted')} ref={c => { this.element = c; }}>
+        <h1 className={styles.title}>
+          Thanks for signing up!
+        </h1>
+        {
+          !this.props.updatedFormSubmitted
```
Comment (zoeabryant) [inline site/pages/home/newsletter-slice/after-sign-up/index.js]: Move to own component in same file

File: site/pages/home/newsletter-slice/index.js [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+    }
+    // !this.state.email_address is a check that we are in before sign up
+    if (inBeforeSignUp && !isValidEmail(data.email_address)) {
+      return this.setState({
+        errorMessage: 'Please enter a valid email address',
+      });
+    }
+    const formDataJSON = JSON.stringify(object);
+    return submitFormFunction({
+      url: 'https://v8pxyg84jj.execute-api.eu-west-1.amazonaws.com/dev/mailing-list',
```
Comment (zoeabryant) [inline site/pages/home/newsletter-slice/index.js]: environment variable

File: site/pages/home/newsletter-slice/index.js [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+    this.state = {
+      newsletterSubmitted: false,
+      email_address: '',
+      errorMessage: '',
+      updatedFormSubmitted: false,
+    };
+    this.submitForm = this.submitForm.bind(this);
+  }
+
+  submitForm(data, method, submitFormFunction = fetchFunction) {
```
Comment (zoeabryant) [inline site/pages/home/newsletter-slice/index.js]: To functions for each form

File: site/pages/home/newsletter-slice/style.css [hunk 13]
```diff
@@ -1,12 +1,50 @@
-@value black, white, daveGreen from "../../../css/_colors.css";
+@value black, white, daveGreen, red, dark-red from "../../../css/_colors.css";
```
Comment (zoeabryant) [inline site/pages/home/newsletter-slice/style.css]: no dark red

File: site/pages/home/newsletter-slice/style.css [hunk 14]
```diff
@@ -1,12 +1,50 @@
...
+  position: relative;
+  padding: 5px 0px 30px 0px;
+}
+
+.buttonSubmitting {
+  opacity: 0.7;
+}
+
+.beforeFormError {
+  outline: 2px solid red;
```
Comment (zoeabryant) [inline site/pages/home/newsletter-slice/style.css]: `border: 2px solid red;`

File: services/mailchimp/utilities/index.js (hunk 2 above)
Comment (AndrewBestbier) [inline services/mailchimp/utilities/index.js]: Done

File: services/mailchimp/utilities/index.js (hunk 3 above)
Comment (AndrewBestbier) [inline services/mailchimp/utilities/index.js]: Use in test setup

File: site/css/_colors.css (hunk 6 above)
Comment (AndrewBestbier) [inline site/css/_colors.css]: Done

File: site/pages/home/newsletter-slice/style.css (hunk 13 above)
Comment (AndrewBestbier) [inline site/pages/home/newsletter-slice/style.css]: Done

File: site/pages/home/newsletter-slice/style.css (hunk 14 above)
Comment (AndrewBestbier) [inline site/pages/home/newsletter-slice/style.css]: Done

File: site/pages/home/newsletter-slice/after-sign-up/index.js (hunk 8 above)
Comment (AndrewBestbier) [inline site/pages/home/newsletter-slice/after-sign-up/index.js]: Done

File: site/pages/home/newsletter-slice/after-sign-up/index.js (hunk 10 above)
Comment (AndrewBestbier) [inline site/pages/home/newsletter-slice/after-sign-up/index.js]: Done

**Pull request merged**